### ISSUE TITLE
1.12 Release progress

### ISSUE THREAD

**Issue by markelog** [jquery/jquery#2787]
See https://github.com/jquery/jquery/issues/2723 for 2.2 version and more info.

Questionable and complicated reverts are the same, but just in case. I don't have access to IE6-7 at the moment, so i'm having trouble reverting https://github.com/jquery/jquery/commit/90d7cc1d8b2ea7ac75f0eacb42439349c9c73278 commit. Help would be appreciated.

Used different revert tactic this time, so in the tests there is places like - 
```js
test( "...."
// and
QUnit.test( "..."
```
etc.

Mostly, i propose for you to check if API is the same as in 2.2 branch.

### Reverted commits

Reason: Behaviour change
Name: Attributes: do not set properties to false when removing booleans
Commit: 5c086c3782459307c44397549fef15a87c8b90c4
Issue: https://github.com/jquery/jquery/issues/1759

Reason: Behaviour change
Name: Core: make isNumeric limited to strings and numbers
Commit: ada073e9acfc9a103b13b9b69d26590e1c834d04
Issue: https://github.com/jquery/jquery/issues/2662

Reason: Follow-up commit for 9df8bd205ab78308c34af70c934c42175e560e86
Name: CSS: Make show/hide/toggle methods a module
Commit: 3842246024475eafdc8a00a7ac4358d06d76cab2
Issue: https://github.com/jquery/jquery/issues/2193
PR: https://github.com/jquery/jquery/pull/2648

Reason: API change (signature)
Name: Effects: Remove additional parameters of easings
Commit: 41a04e59df5beac4ca61b3b5e3a6b078b6aee1ac
Issue: https://github.com/jquery/jquery/issues/2367
PR: https://github.com/jquery/jquery/pull/2376

Reason: API change (new method)
Name: Deferred: add .catch handler
Commit: ef77f83db86b806425203d1e5ecbcc119d7b9c32
Issue: https://github.com/jquery/jquery/issues/2102

Reason: Follow-up commit for ef30bdf4f1b361b6f2e99f5b7233b419f95a62f2
Name: Ajax: Remove remnants of the load event alias handling
Commit: 38bd5feddeca3acdd740300f370450621d4e2731

Reason: API change (removal of public methods)
Name: Event: remove deprecated event aliases
Commit: ef30bdf4f1b361b6f2e99f5b7233b419f95a62f2
Issue: https://github.com/jquery/jquery/issues/2286
PR: https://github.com/jquery/jquery/pull/2287

Reason: Behaviour change
Name: CSS: Ignore the CSS cascade in show()/hide()/etc.
Commit: 9df8bd205ab78308c34af70c934c42175e560e86
Issue: https://github.com/jquery/jquery/issues/1767
Issue: https://github.com/jquery/jquery/issues/2071
PR: https://github.com/jquery/jquery/pull/2180

Reason: Behaviour change
Name: Offset: allow offset setter to throw for disconnected elements 
Commit: dc49f62f22a90ba95bbd1b659bfdae905adb4dd8
Issue: https://github.com/jquery/jquery/issues/2114

Reason: Behaviour change
Name: Offset: remove ownerDocument check in offset getter
Commit: a2386a8250c82e27680d21f7f6fd1eb9a8f7e938
Issue: https://github.com/jquery/jquery/issues/2115

Reason: Behaviour change
Name: Offset: Fix .offset() to correctly work with ShadowDOM
Commit: d4dd548acaf049d8e4dca9e4b879a51f26bb3d27
Issue: https://github.com/jquery/jquery/issues/1784
PR: https://github.com/jquery/jquery/pull/2043

Reason: Behaviour change
Name: Data: camelCasing should not ignore case
Commit: cf16f860b05b7bffeb2382a0a55bb85c69b2abb5
Issue: https://github.com/jquery/jquery/issues/2070

Reason: Behaviour change
Name: Data: always camelCase keys in .data()
Commit: 0204c3089e7beee0306594605cc64d1e050ecd07
Issue: https://github.com/jquery/jquery/issues/2257

Reason: Behaviour change
Name: Data: do not include digits when camelCasing
Commit: a254f22d7bb5b072a82087a448130ffadd70d698
Issue: https://github.com/jquery/jquery/issues/1751

Reason: Behaviour change
Name: Data: do not include digits when camelCasing
Commit: a254f22d7bb5b072a82087a448130ffadd70d698
Issue: https://github.com/jquery/jquery/issues/1751

Reason: API change (removal of public methods)
Name: Ajax: remove deprecated extensions from ajax promise
Commit: 9368a8cd8ca81902b8359188668973e489fca6db
Issue: https://github.com/jquery/jquery/issues/2084
PR: https://github.com/jquery/jquery/pull/2092

Reason: IE6-7 compliance 
Name: Manipulation: don't auto-insert tbody
Commit: 939e755163c9c9749fdd39ae9e60de7cb41eeb2c
Issue: https://github.com/jquery/jquery/issues/1835
PR: https://github.com/jquery/jquery/pull/2021

Reason: Behaviour change
Name: Callbacks: Don't abort execution on .lock()
Commit: fc7477f4927100f23f83c22b5142a42d1eac8502
Issue: https://github.com/jquery/jquery/issues/1990
PR: https://github.com/jquery/jquery/pull/1991

Reason: IE6-7 compliance
Name: CSS: simplify hack of css getter for the computed values
Commit: dac716ca65a83bab2449ffa35c43b4b52e26a0c1
Issue: https://github.com/jquery/jquery/issues/1906

Reason: Follow-up commit for 90d7cc1d8b2ea7ac75f0eacb42439349c9c73278 
Name: CSS: simplify "defaultDisplay" module
Commit: d9d8906cfdd05a6163b0a6791f7ab4433951eacd

Reason: Follow-up commit for 90d7cc1d8b2ea7ac75f0eacb42439349c9c73278 
Name: Traversing: simplify jQuery#contents method
Commit: 7230df15ac612e0e5a08ed7ffc02f6700b2ec898

Reason: Behaviour change
Name: Manipulation: make wrapAll funarg execute only once
Commit: 3dcee021603d5c25cd429cb25c2de5046044ca4c
Issue: https://github.com/jquery/jquery/issues/1843

Reason: Behaviour change
Name: Core: Throw an error on $("#") rather than returning 0-length collection
Commit: 93f95c966355f02b5695f50523f88687eadb4c81
Issue: https://github.com/jquery/jquery/pull/1682

Reason: API change (removal of public properties) 
Name: Core: Remove deprecated context and selector properties
Commit: e2ec5da2a7f1b3de1f9dc7e2c01f27cd23714e60
Issue: https://github.com/jquery/jquery/issues/1908
PR: https://github.com/jquery/jquery/pull/2000

Reason: Follow-up commit for 06a45406966ee8cde31c4f128d7ee68d727880c1
Name: Effects: Improve raf logic
Commit: 9dc29a2b130e6bbcdbcaf8fdc1433a41e6b7a585

Reason: Behaviour change
Name: Effects: Reintroduce use of requestAnimationFrame
Commit: 06a45406966ee8cde31c4f128d7ee68d727880c1
Issue: trac-15147

Reason: Behaviour change
Name: Ajax: Always use script injection in globalEval
Commit: 37f0f7f42cd158e36665066d88708ca4aa0713ac
Issue: trac-14757

Reason: Old Opera compliance
Name: Ajax: simplify one ajax call and add explanatory comment
Commit: 4b3e63066dd480d07b9ce8057cb0c02b8ad7e990

Reason: IE6-7 compliance
Name: Manipulation: simplification of manipulation wrappers
Commit: 3e8712062d92feea8723e6185c47a2cc98e0e7af
PR: https://github.com/jquery/jquery/pull/2032

Reason: API change (different return value)
Name: Core: Return empty array instead of null for parseHTML("")
Commit: 61bb61279ce51af189d40162dc62e44f04826051
Issue: https://github.com/jquery/jquery/issues/1997
PR: https://github.com/jquery/jquery/pull/1998

Reason: API change (signature change {feature})
Name: Wrap: Support .unwrap( selector) for selective unwrapping
Commit: e38138af6ac3fa850a092adf09366a68850189a8
Issue: https://github.com/jquery/jquery/issues/1744
PR: https://github.com/jquery/jquery/pull/2003

--
Also removed effects tests for jQuery.Animate and jQuery.Tween by cherry-picking
https://github.com/jquery/jquery/commit/a5864aefdfadfee16522456c82139fa10360d8fc

/cc @jquery/core

**Comment by mgol** [comment]
@markelog This should have the 1.12/2.2 label, shouldn't it? Similarly to #2723.

**Comment by markelog** [comment]
Added, thank yoiu

**Comment by timmywil** [comment]
1.12-stable has been updated. There are some failures in ie6, but it's looking very green now. Also verified that 1.12 has the same API additions as 2.2; I'm gonna rub my eyes for a while now.
